Mastodon/Protocol/Provider/DataSourceFacade+Profile.swift

```diff
@@ -12,23 +12,37 @@ import MastodonSDK
 
 extension DataSourceFacade {
     
+    @MainActor
     static func coordinateToProfileScene(
         provider: DataSourceProvider & AuthContextProvider,
         target: StatusTarget,
-        status: ManagedObjectRecord<Status>
+        status: MastodonStatus
     ) async {
-        let _redirectRecord = await DataSourceFacade.author(
-            managedObjectContext: provider.context.managedObjectContext,
-            status: status,
-            target: target
-        )
+        let acct: String
+        switch target {
+        case .status:
+            acct = status.reblog?.entity.account.acct ?? status.entity.account.acct
+        case .reblog:
+            acct = status.entity.account.acct
+        }
+        
+        provider.coordinator.showLoading()
+        
+        let _redirectRecord = try? await Mastodon.API.Account.lookupAccount(
+            session: .shared,
+            domain: provider.authContext.mastodonAuthenticationBox.domain,
+            query: .init(acct: acct),
+            authorization: provider.authContext.mastodonAuthenticationBox.userAuthorization
+        ).singleOutput().value
+                
         guard let redirectRecord = _redirectRecord else {
             assertionFailure()
+            provider.coordinator.hideLoading()
             return
         }
         await coordinateToProfileScene(
```

Comment:
I noticed that the spinner isn't hidden again, so maybe add a `provider.coordinator.hideLoading()` here?